libPhoneNumber/NBPhoneNumberUtil.m

```diff
@@ -3496,15 +3496,26 @@ - (CTTelephonyNetworkInfo *)telephonyNetworkInfo {
 }
 
 - (NSString *)countryCodeByCarrier {
-  NSString *isoCode = [[self.telephonyNetworkInfo subscriberCellularProvider] isoCountryCode];
-
-  // The 2nd part of the if is working around an iOS 7 bug
-  // If the SIM card is missing, iOS 7 returns an empty string instead of nil
-  if (isoCode.length == 0) {
-    isoCode = NB_UNKNOWN_REGION;
-  }
-
-  return isoCode;
+#if __IPHONE_OS_VERSION_MIN_REQUIRED >= 120000
+    NSDictionary<NSString *, CTCarrier *> *serviceSubscriberCellularProviders = [self.telephonyNetworkInfo serviceSubscriberCellularProviders];
+    CTCarrier *carrier = serviceSubscriberCellularProviders.allValues.firstObject;
+    
+    if (nil==carrier) {
```

Comment:
This project uses Google Objective-C style guide, we generally do: if (carrier == nil) { ... }

If you want, you can also run clang-format -style='Google' to format the code.

Reply:
Sorry @FONickReichard - that's from my habit if always putting `nil` first (to ensure we don't accidentally assign nil in the if check! 😉

Reply:
Np! Thanks for the feedback. Noted to follow the Google Objective-C style guide.